src/kimmdy/topology/topology.py

```diff
@@ -551,33 +594,114 @@ def __init__(
         molecules = get_top_section(top, "molecules")
         if molecules is None:
             raise ValueError("molecules not found in top file")
-        self.molecules = {l[0]: l[1] for l in molecules}
+        self.molecules = [(l[0], l[1]) for l in molecules]
 
         self.ff = FF(top)
         self._parse_molecules()
         self.parametrizer = parametrizer
         self.needs_parameterization = False
 
-        # link atoms, bonds etc. to the main moleculeype, assumed to be the first
-        # "moleculetype_0"
-        self.main_molecule_ix = 0
-        self.main_molecule_name = list(self.moleculetypes.keys())[self.main_molecule_ix]
-        self.main_molecule = self.moleculetypes[self.main_molecule_name]
-        self.atoms = self.main_molecule.atoms
-        self.bonds = self.main_molecule.bonds
-        self.angles = self.main_molecule.angles
-        self.proper_dihedrals = self.main_molecule.proper_dihedrals
+        self._merge_moleculetypes()
+        self._link_atomics()
+
+    def _link_atomics(self):
+        """
+        link atoms, bonds etc. to the main moleculeype, REACTIVE_MOLECULEYPE
+        """
+        self.reactive_molecule = self.moleculetypes[REACTIVE_MOLECULEYPE]
+        self.atoms = self.reactive_molecule.atoms
+        self.bonds = self.reactive_molecule.bonds
+        self.angles = self.reactive_molecule.angles
+        self.proper_dihedrals = self.reactive_molecule.proper_dihedrals
         self.improper_dihedrals = self.moleculetypes[
-            self.main_molecule_name
+            REACTIVE_MOLECULEYPE
         ].improper_dihedrals
-        self.pairs = self.main_molecule.pairs
+        self.pairs = self.reactive_molecule.pairs
         self.position_restraints = self.moleculetypes[
-            self.main_molecule_name
+            REACTIVE_MOLECULEYPE
         ].position_restraints
         self.dihedral_restraints = self.moleculetypes[
-            self.main_molecule_name
+            REACTIVE_MOLECULEYPE
         ].dihedral_restraints
-        self.radicals = self.main_molecule.radicals
+        self.radicals = self.reactive_molecule.radicals
+
+    def _extract_mergable_molecules(self):
```

Comment:
I think this behavior is good but not clear to a user who hasn't thought about this problem and could result in kimmdy doing something that the user did not expect.
What do you think about:
- exposing the choice of mergable molecules in the config
- adding more logging.info to this step
- adding a "Design Choices" section to the wiki that explains unintuitive behavior of kimmdy?

Reply:
There is a `guide/explanation/topology` section on this, but it can be fleshed out more in the future and linked from important places such as the getting started.

The feature is built with configuration in mind (hence the predicate function), but for now the defaults are equivalent for any system that would have worked previously with kimmdy and just make additional systems also work. Right now there is no real usecase for configuring this differently, but we can keep the option open.

Reply:
The only other system I could think of was my hydrolysis, where I would have to add water to the reactive type, but this requires some special handling of the `settles` directive, since this is a topology section unique to a solvent and required to be defined only once and in a certain position in the topology. So until this comes up I'd rather keep it consistent for now instead of handling this edge case before needed.

Reply:
that's a good example. I will make it configurable already.